superset/utils/log.py

```diff
@@ -133,8 +154,29 @@ def log_context(  # pylint: disable=too-many-locals
             records=records,
             dashboard_id=dashboard_id,
             slice_id=slice_id,
-            duration_ms=round((time.time() - start_time) * 1000),
+            duration_ms=int(duration.total_seconds() * 1000),
             referrer=referrer,
+            **kwargs,
+        )
+
+    @contextmanager
+    def log_context(  # pylint: disable=too-many-locals
+        self, action: str, object_ref: Optional[str] = None, log_to_statsd: bool = True,
+    ) -> Iterator[Callable[..., None]]:
+        """
+        Log an event with additional information from the request context.
+        :param action: a name to identify the event
+        :param object_ref: reference to the Python object that triggered this action
+        :param log_to_statsd: whether to update statsd counter for the action
+        """
+        payload_override = {}
+        start = datetime.now()
+        # yield a helper to add additional payload
+        yield lambda **kwargs: payload_override.update(kwargs)
+        duration = datetime.now() - start
+
+        self.log_with_context(
+            action, duration, object_ref, log_to_statsd, payload_override
```

Comment:
```suggestion
            action, duration, object_ref, log_to_statsd, **payload_override
```

See my comment above.